netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -109,6 +109,9 @@
   @VisibleForTesting
   static final long GRACEFUL_SHUTDOWN_PING = 0x97ACEF001L;
   private static final long GRACEFUL_SHUTDOWN_PING_TIMEOUT_NANOS = TimeUnit.SECONDS.toNanos(10);
+  /** Temporary workaround for #8674. Fine to delete after v1.45 release, and maybe earlier. */
+  private static final boolean DISABLE_CONNECTION_HEADER_CHECK = Boolean.parseBoolean(
+      System.getProperty("io.grpc.netty.disableConnectionHeaderCheck", "false"));
```

Comment:
This is equivalent to `Boolean.getBoolean("io.grpc.netty.disableConnectionHeaderCheck")`

Reply:
While more convenient, that API is much less clear and we tend to do this approach (I don't see a single instance of `Boolean.getBoolean()` in the codebase). The `"false"` argument happens to be unnecessary, but I felt like leaving it for clarity.

Reply:
We don't have `Boolean.getBoolean()` in the codebase because all the occurrences of `Boolean.parseBoolean()` were from env, not System properties, whereas `Boolean.getBoolean()` does not parse env variable.

Reply:
We do have _some_ such usages.
https://github.com/grpc/grpc-java/blob/d4fa0ecc07495097453b0a2848765f076b9e714c/core/src/main/java/io/grpc/internal/ManagedChannelOrphanWrapper.java#L75
https://github.com/grpc/grpc-java/blob/876f56e2ea32b9dfe13e96dc37111ef216f796ae/api/src/main/java/io/grpc/Status.java#L235
https://github.com/grpc/grpc-java/blob/d4fa0ecc07495097453b0a2848765f076b9e714c/rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java#L95
https://github.com/grpc/grpc-java/blob/c1e19af86dea0b9a8725969a95e116029397ad4d/grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java#L95